src/lib/support/tests/TestTestPersistentStorageDelegate.cpp

```diff
@@ -107,66 +109,68 @@ void TestBasicApi(nlTestSuite * inSuite, void * inContext)
 
     // Read them back
 
+    uint8_t all_zeroes[sizeof(buf)];
+    memset(&all_zeroes[0], 0, sizeof(all_zeroes));
+
     memset(&buf[0], 0, sizeof(buf));
-    size = sizeof(buf);
+    size = actualSizeOfBuf;
     err  = storage.SyncGetKeyValue("key2", &buf[0], size);
     NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
     NL_TEST_ASSERT(inSuite, size == strlen(kStringValue2));
+    NL_TEST_ASSERT(inSuite, size < actualSizeOfBuf);
     NL_TEST_ASSERT(inSuite, 0 == memcmp(&buf[0], kStringValue2, strlen(kStringValue2)));
+    // Make sure that there was no buffer overflow during SyncGetKeyValue
+    NL_TEST_ASSERT(inSuite, 0 == memcmp(&buf[size], &all_zeroes[0], sizeof(buf) - size));
 
     memset(&buf[0], 0, sizeof(buf));
-    size = sizeof(buf);
+    size = actualSizeOfBuf;
     err  = storage.SyncGetKeyValue("key3", &buf[0], size);
     NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
     NL_TEST_ASSERT(inSuite, size == strlen(kStringValue3));
+    NL_TEST_ASSERT(inSuite, size < actualSizeOfBuf);
```

Comment:
```suggestion
    NL_TEST_ASSERT(inSuite, size <= actualSizeOfBuf);
```

Reply:
We intentionally are checking that this is smaller than the actual size of the buffer of `buf`

Reply:
But why?